src/body_operations.cpp

```diff
@@ -160,9 +160,11 @@ void bodies::computeBoundingSphere(const std::vector<const bodies::Body*>& bodie
   unsigned int vertex_count = 0;
   for (auto body : bodies)
   {
-    const bodies::ConvexMesh* conv = dynamic_cast<const bodies::ConvexMesh*>(body);
-    if (!conv)
+    if (body->type != MESH)
```

Comment:
First test if `body != nullptr`. The old check would rule such cases out, whether the suggested change would segfault.